src/store/channels-list/saga.ts

```diff
@@ -428,12 +432,17 @@ export function* otherUserJoinedChannel(roomId: string, user: User) {
     return;
   }
 
-  // TODO: Fetch user from zOS if we don't know about them yet
+  if (user.userId === user.matrixId) {
+    user = yield select(userByMatrixId, user.matrixId);
+  }
+
   if (!channel.otherMembers.includes(user.userId)) {
+    const otherMembers = [...channel.otherMembers, user];
     yield put(
       receiveChannel({
         id: channel.id,
-        otherMembers: [...channel.otherMembers, user],
+        isOneOnOne: channel.isChannel === true ? false : otherMembers.length === 1,
```

Comment:
This is still incorrect. A Conversation is not a "One on One" if it _ever_ had more than two people in it. Here you'd be setting old group conversations to one on ones if everyone else left.